src/main/java/io/appium/java_client/android/AndroidMobileOptions.java

```diff
@@ -0,0 +1,1716 @@
+        return amend(AndroidMobileCapabilityType.AVD_LAUNCH_TIMEOUT, ms);
+    }
+
+    /**
+     * Get the wait time for an avd to launch and connect to ADB.
+     *
+     * @return Duration to wait for an avd to launch and connect to ADB.
+     * @see AndroidMobileCapabilityType#AVD_LAUNCH_TIMEOUT
+     */
+    public Duration getAvdLaunchTimeout() {
+        return (Duration) getCapability(AndroidMobileCapabilityType.AVD_LAUNCH_TIMEOUT);
+    }
+
+    /**
+     * Set the wait time for an avd to finish its boot animations.
+     *
+     * @param ms is the number of milliseconds to wait for an avd to finish its boot animations.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#AVD_READY_TIMEOUT
+     */
+    public AndroidMobileOptions setAvdReadyTimeout(Duration ms) {
+        return amend(AndroidMobileCapabilityType.AVD_READY_TIMEOUT, ms);
+    }
+
+    /**
+     * Get the the wait time for an avd to finish its boot animations.
+     *
+     * @return Duration to wait for an avd to finish its boot animations.
+     * @see AndroidMobileCapabilityType#AVD_READY_TIMEOUT
+     */
+    public Duration getAvdReadyTimeout() {
+        return (Duration) getCapability(AndroidMobileCapabilityType.AVD_READY_TIMEOUT);
+    }
+
+    /**
+     * Set the Android build-tools version to be something different than the default.
+     *
+     * @param version is the version of Android build-tools to use.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#BUILD_TOOLS_VERSION
+     */
+    public AndroidMobileOptions setBuildToolsVersion(String version) {
+        return amend(AndroidMobileCapabilityType.BUILD_TOOLS_VERSION, version);
+    }
+
+    /**
+     * Get the Android build-tools version to be something different than the default.
+     *
+     * @return String of the Android build-tools version to be something different than the default.
+     * @see AndroidMobileCapabilityType#BUILD_TOOLS_VERSION
+     */
+    public String getBuildToolsVersion() {
+        return (String) getCapability(AndroidMobileCapabilityType.BUILD_TOOLS_VERSION);
+    }
+
+    /**
+     * Set the arguments to be passed to the chromedriver binary when it's run by Appium..
+     *
+     * @param args is the List of arguments to be passed to the chromedriver.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#CHROMEDRIVER_ARGS
+     */
+    public AndroidMobileOptions setChromedriverArgs(List<String> args) {
+        return amend(AndroidMobileCapabilityType.CHROMEDRIVER_ARGS, args);
+    }
+
+    /**
+     * Get the arguments to be passed to the chromedriver binary when it's run by Appium..
+     *
+     * @return list of String arguments to be passed to the chromedriver binary when it's run by Appium.
+     * @see AndroidMobileCapabilityType#CHROMEDRIVER_ARGS
+     */
+    public Object getChromedriverArgs() {
```

Comment:
why Object?

Reply:
It returns correctly as a List, and I can't cast the result into one. Is there a better way here? Java typing isn't my strong suit. :)

Reply:
if the `incoming` argument type is `List<String>` then I would expect this one to be the same

Reply:
> It returns correctly as a List, and I can't cast the result into one

Why it is not possible to cast?

Reply:
Well, IntelliJ was complaining before, and it isn't now. So I'll update :)